model/host/volume.go

```diff
@@ -157,6 +170,23 @@ func FindTotalVolumeSizeByUser(user string) (int, error) {
 	return out[0].TotalVolumeSize, nil
 }
 
+func FindVolumesWithNoExpirationToExtend() ([]Volume, error) {
+	query := bson.M{
+		VolumeNoExpirationKey: true,
+		VolumeHostKey:         bson.M{"$exists": false},
```

Comment:
I'm still confused about why this is part of the query. Do attached hosts not expire? What if we unattach a volume an unexpirable volume and now the expiration we have is really out of date?

Reply:
Attached volumes don't expire (they're excluded from [the query](https://github.com/evergreen-ci/evergreen/blob/9d5c9206f7ed0b12474baf2111a651ac1352044c/model/host/volume.go#L94))

The expiration is extended when [a volume is detached](https://github.com/evergreen-ci/evergreen/blob/9d5c9206f7ed0b12474baf2111a651ac1352044c/cloud/ec2.go#L1268) or the [host it's attached to is terminated](https://github.com/evergreen-ci/evergreen/blob/9d5c9206f7ed0b12474baf2111a651ac1352044c/cloud/ec2.go#L972)